playwright/integrations/unauthenticated-chat-with-reconnect.spec.ts

```diff
@@ -0,0 +1,78 @@
+            }),
+            page.waitForRequest(request => {
+                return request.url().includes(OmnichannelEndpoints.LiveChatSessionInitPath);
+            }),
+            page.waitForResponse(response => {
+                return response.url().includes(OmnichannelEndpoints.LiveChatSessionInitPath);
+            }),
+            page.waitForRequest(request => {
+                return request.url().includes(OmnichannelEndpoints.LiveChatReConnect);
+            }),
+            page.waitForResponse(response => {
+                return response.url().includes(OmnichannelEndpoints.LiveChatReConnect);
+            }),
+            await page.evaluate(async ({ omnichannelConfig }) => {
+                const { OmnichannelChatSDK_1: OmnichannelChatSDK } = window;
+                const chatSDKConfig = {
+                    chatReconnect: {
+                        disable: false,
+                    },
+                }
+                const chatSDK = new OmnichannelChatSDK.default(omnichannelConfig, chatSDKConfig);
+
+                const runtimeContext = {};
+
+                await chatSDK.initialize();
+
+                const params = {
+                    reconnectId: omnichannelConfig.reconnectId
+                };
+
+                const chatReconnectContext = await chatSDK.getChatReconnectContext(params);
+
+                runtimeContext.reconnectId = chatReconnectContext.reconnectId;
+                runtimeContext.redirectURL = chatReconnectContext.redirectURL;
+                runtimeContext.requestId = chatSDK.requestId;
+
+                await chatSDK.startChat();
+
+                await chatSDK.endChat();
+
+                return runtimeContext;
+            }, { omnichannelConfig })
+        ]);
+
+        const { reconnectId, redirectURL, requestId } = runtimeContext;
+        const chatTokenRequestUrl = `${omnichannelConfig.orgUrl}/${OmnichannelEndpoints.LiveChatv2GetChatTokenPath}/${omnichannelConfig.orgId}/${omnichannelConfig.widgetId}/${requestId}?channelId=lcw`;
+        const sessionInitRequestUrl = `${omnichannelConfig.orgUrl}/${OmnichannelEndpoints.LiveChatSessionInitPath}/${omnichannelConfig.orgId}/${omnichannelConfig.widgetId}/${requestId}?channelId=lcw`;
+        const reconnectRequestUrl = `${omnichannelConfig.orgUrl}/${OmnichannelEndpoints.LiveChatReConnect}/${omnichannelConfig.orgId}/${omnichannelConfig.widgetId}/${omnichannelConfig.reconnectId}`;
+
+        expect(chatTokenRequest.url() === chatTokenRequestUrl).toBe(true);
+        expect(chatTokenResponse.status()).toBe(200);
+        expect(sessionInitRequest.url() === sessionInitRequestUrl).toBe(true);
+        expect(sessionInitResponse.status()).toBe(200);
+        expect(reconnectRequest.url() === reconnectRequestUrl).toBe(true);
+        expect(reconnectResponse.status()).toBe(200);
+
+        expect(reconnectId).toBe(null);
+        expect(redirectURL).toBe('https://www.microsoft.com/');
```

Comment:
We should just ensure that `redirectURL` should be `null` and it should be good enough. Otherwise, the test will fail if the reconnectURL was not always set to this value. It'd make the test case more scalable and flexible